### PR TITLE
Improve spec file compatibility [2/2]

### PR DESCRIPTION
**Pull request by schmittjoseph**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

This PR introduces several changes to spec files that add missing provides. This is the accompanying PR to the[ tooling changes](https://github.com/microsoft/CBL-Mariner/pull/141). Both will be merged at the same time once both are approved.

###### Change Log  

- Add support for multiple macros through `mariner-rpm-macros`.
- Include additional packages in the worker chroot, such as `python`, as specs are now parsed in this environment. Specs can rely on being able to call these programs to configure their provides or packages., 
- Switch from `pkg-config` to `pkgconf`. `pkg-config` is no longer being developed, switch to an actively maintained drop-in replacement that has additional features.
- Fix toolchain specs such as `openjdk8` to have all of their build requirements present in the toolchain environment before being built. This issue was resulting in packages being rebuilt when not needed.
- Add missing or synonymous provides to various packages.
- Fix ambiguous specs that did not build with a specific version of `python`. Now that `python` is in the worker chroot, specs should explicitly state if they wish to build with `python2` or `python3` to ensure they are built correctly.
- Support [DWARF](https://en.wikipedia.org/wiki/DWARF) compression of debug symbols.
- Replace lua spec with Fedora version, which provides more robust subpackages, file structure, and provides.

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Local builds.

### CODE REVIEW

**Comment by schmittjoseph** [comment]
Manifest checks failing is due to specs now leveraging `%include`. The [tooling pr](https://github.com/microsoft/CBL-Mariner/pull/141) introduces changes to manifest checks to support this. This CI failure will be resolved once the prs are in.

File: SPECS/systemtap/systemtap.spec [hunk 1]
```diff
@@ -1,3 +1,5 @@
+%define _unpackaged_files_terminate_build 0
```
Comment (hbeberman) [inline SPECS/systemtap/systemtap.spec:1]: :(

**Review by hbeberman — APPROVED** [review]

File: SPECS/lldb/lldb.spec [hunk 2]
```diff
@@ -58,7 +58,10 @@ cmake -DCMAKE_INSTALL_PREFIX=/usr           \
       -DLLDB_PATH_TO_CLANG_BUILD=%{_prefix} \
       -DLLVM_DIR=/usr/lib/cmake/llvm        \
       -DLLVM_BUILD_LLVM_DYLIB=ON ..         \
-      -DLLDB_DISABLE_LIBEDIT:BOOL=ON
+      -DLLDB_DISABLE_LIBEDIT:BOOL=ON        \
+      -DPYTHON_EXECUTABLE:STRING=%{__python2} \
```
[the remark this answers is not in the document]
Comment (schmittjoseph) [reply SPECS/lldb/lldb.spec:62]: This spec was originally building with python2 so I'm preserving this behavior. If we want to update this spec to use python3 I think that should happen in a separate PR.

File: SPECS/openscap/openscap.spec [hunk 3]
```diff
@@ -58,6 +58,9 @@ mkdir build
 cd build
 %cmake -DENABLE_PERL=ON \
        -DENABLE_SCE=ON \
+       -DPYTHON_EXECUTABLE:STRING=%{__python2} \
```
[the remark this answers is not in the document]
Comment (schmittjoseph) [reply SPECS/openscap/openscap.spec:61]: This spec was originally building with python2 so I'm preserving this behavior. If we want to update this spec to use python3 I think that should happen in a separate PR.

File: SPECS/dwz/dwz.spec [hunk 4]
```diff
@@ -0,0 +1,140 @@
+Summary:        DWARF optimization and duplicate removal tool
+Name:           dwz
+Version:        0.13
+Release:        4%{?dist}
+License:        GPLv2+ and GPLv3+
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            https://sourceware.org/dwz
+Source:         https://sourceware.org/ftp/dwz/releases/%{name}-%{version}.tar.xz
+BuildRequires:  gcc, elfutils-libelf-devel
```
Comment (PawelWMS) [inline SPECS/dwz/dwz.spec:10]: Nit: one per line.

File: SPECS/libarchive/libarchive.spec [hunk 5]
```diff
@@ -76,7 +78,7 @@ make %{?_smp_mflags} check
 -   Fixed "Source0" and "URL" tags.
 -   License verified.
 *   Tue Sep 03 2019 Mateusz Malisz <mamalisz@microsoft.com> 3.3.3-2
--   Initial CBL-Mariner import from Photon (license: Apache2).
+-   Initial CBL-Mariner import from Photon (license: dual Apache2/GPL2).
```
Comment (PawelWMS) [inline SPECS/libarchive/libarchive.spec:81]: Previous version was actually correct.

File: SPECS/pkgconf/pkgconf.spec [hunk 6]
```diff
@@ -0,0 +1,351 @@
...
+%files pkg-config
+%{_bindir}/pkg-config
+%{_bindir}/%{_target_platform}-pkg-config
+%{_mandir}/man1/pkg-config.1*
+%dir %{_libdir}/pkgconfig
+%dir %{_datadir}/pkgconfig
+%endif
+
+%changelog
+* Mon Sep 28 2020 Joe Schmitt <joschmit@microsoft.com> - 1.7.0-2
```
Comment (PawelWMS) [inline SPECS/pkgconf/pkgconf.spec:187]: Nit: release different from the one in the `Release` tag.

File: SPECS/libarchive/libarchive.spec (hunk 5 above)
Comment (schmittjoseph) [reply SPECS/libarchive/libarchive.spec:81]: Will update in a separate PR. good catch

**Pull request merged**